test-validator/src/lib.rs

```diff
@@ -312,6 +314,38 @@ impl TestValidatorGenesis {
         Ok(self)
     }
 
+    pub fn clone_upgradeable_programs<T>(
+        &mut self,
+        addresses: T,
+        rpc_client: &RpcClient,
+    ) -> Result<&mut Self, String>
+    where
+        T: IntoIterator<Item = Pubkey> + Clone,
+    {
+        self.clone_accounts(addresses.clone(), rpc_client, false)?;
+
+        let mut programdata_addresses: HashSet<Pubkey> = HashSet::new();
+        for address in addresses {
+            let account_shared = self.accounts.get(&address).unwrap();
+            let account = Account::from(account_shared.clone());
```

Comment:
Rather than copy this account here, you can call `data()` on `account_shared` to get the data bytes when deserializing

Reply:
I've just realized that there's also `deserialize_data()` so the bincode dep can be removed

Reply:
ah, very slick!